apps/estuary/test/integration/estuary_test.exs

```diff
@@ -0,0 +1,134 @@
+    ]
+
+    actual_columns =
+      "DESCRIBE #{@event_stream_table_name}"
+      |> Prestige.execute()
+      |> Prestige.prefetch()
+
+    assert expected_columns == actual_columns
+  end
+
+  test "should persist event to the event_stream table" do
+    produce_event(
+      @event_stream_topic,
+      %{
+        "author" => "reaper",
+        "create_ts" => 5,
+        "data" => "some data for reaper",
+        "type" => "some type for reaper"
+      }
+      |> event_struct()
+    )
+
+    expected_value = [["reaper", 5, "some data for reaper", "some type for reaper"]]
+
+    eventually(fn ->
+      actual_value =
+        "'reaper'"
+        |> EventTableHelper.select_table_data()
+
+      assert expected_value == actual_value
+    end)
+
+    EventTableHelper.delete_table_data()
+  end
+
+  test "should persist batch of events to the event stream" do
+    produce_event(@event_stream_topic, [
+      %{
+        "author" => "forklift",
+        "create_ts" => 1,
+        "data" => "some data for forklift",
+        "type" => "some type for forklift"
+      }
+      |> event_struct(),
+      %{
+        "author" => "valkyrie",
+        "create_ts" => 2,
+        "data" => "some data for valkyrie",
+        "type" => "some type for valkyrie"
+      }
+      |> event_struct()
+    ])
+
+    expected_value = [
+      ["forklift", 1, "some data for forklift", "some type for forklift"],
+      ["valkyrie", 2, "some data for valkyrie", "some type for valkyrie"]
+    ]
+
+    eventually(fn ->
+      actual_value =
+        "'forklift', 'valkyrie'"
+        |> EventTableHelper.select_table_data()
+
+      assert expected_value == actual_value
+    end)
+
+    EventTableHelper.delete_table_data()
+  end
+
+  test "should send event to the dlq if it is not a properly formatted event for Jason decoding" do
+    produce_event(@event_stream_topic, {"some_improper_event_key", "some_improper_event_value"})
+
+    expected_value = [
+      "key: \\\"some_improper_event_key\\\"",
+      "value: \\\"some_improper_event_value\\\""
+    ]
+
+    eventually(fn ->
+      assert event_value(expected_value)
+    end)
+  end
+
+  test "should send event to the dlq if it is properly formatted, but doesn't have the right keys" do
+    produce_event(@event_stream_topic, ~s({"some_bad_key": "some_bad_value"}))
+    expected_value = ["{\\\\\\\"some_bad_key\\\\\\\": \\\\\\\"some_bad_value\\\\\\\"}"]
+
+    eventually(fn ->
+      assert event_value(expected_value)
+    end)
+  end
+
+  defp produce_event(topic, payload) do
+    Elsa.produce(@elsa_endpoint, topic, payload)
+  end
+
+  defp event_struct(event_data) do
+    ~s({
+      "__brook_struct__":"Elixir.Brook.Event",
+      "__struct__":"Elixir.SmartCity.Dataset",
+      "author":"#{event_data["author"]}",
+      "create_ts":"#{event_data["create_ts"]}",
+      "data":"#{event_data["data"]}",
+      "forwarded":false,
+      "type":"#{event_data["type"]}"
+      })
+  end
+
+  defp event_value(expected_value) do
```

Comment:
I had 0 idea what this function was doing based on its name.  Something along the lines of `event_exists_in_dlq?` might be clearer

Reply:
agreed. I should probably of put wip on this pull request. We started refactoring and that was the quick name we made at first. It is now say is_in_dlq.